utils/s3/client.cc

```diff
@@ -98,6 +98,191 @@ future<> client::delete_object(sstring object_name) {
     co_await _http.make_request(std::move(req), ignore_reply, http::reply::status_type::no_content);
 }
 
+class client::upload_sink : public data_sink_impl {
+    // "Each part must be at least 5 MB in size, except the last part."
+    // https://docs.aws.amazon.com/AmazonS3/latest/API/API_UploadPart.html
+    static constexpr size_t minimum_part_size = 5 << 20;
+    static constexpr int flush_concurrency = 3;
+
```

Comment:
this seems like write behind for files. of course, network concurrency cannot be treated the same as disk one, but perhaps make it configurable?

Reply:
Will do, yes

Reply:
#13021